setup.py

```diff
@@ -65,6 +63,7 @@
             "pathoc = pathod.pathoc_cmdline:go_pathoc"
         ]
     },
+    python_requires='>=3.8',
```

Comment:
@Kriechi: Is there a reason why we didn't do this before?

See https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires.

This is checked since pip 9, released 2016-11-02.
Do we want to 1) forcibly downgrade pip (which may bring its own issues) to make installs succeed, or 2) drop the test?

Reply:
Previously we still wanted to "support" py27 and py35 to at least try and install the package, and then tell them with our own  custom message to upgrade.
Back then, pip was still fairly new, and we wanted to supported older versions as well.
Now that pip 9 is almost 4 years old, I think it is safe to bump this, and get rid of our own checks around this.